src/js/components/editor/Column.vue

```diff
@@ -0,0 +1,288 @@
+     * 
+     * @param {KeyboardEvent} e triggered event
+     */
+    processMetaInput(e) {
+      switch(e.keyCode) {
+
+        /* disallowed actions */
+        case KEYS.ALPHA.I: // meta + i = italic
+        case KEYS.ALPHA.B: // meta + b = bold
+          e.preventDefault()
+          break;
+      }
+    },
+
+    /**
+     * Synchronizes the model with a new line
+     * 
+     * @param {KeyboardEvent} e triggered event
+     */
+    insertLineAtSelection(e, { line, node }) {
+
+      // whereas the line model still reflects the previous state
+      // the node should be the newly inserted bit.
+      let newLine = this.column.splitLine(line, (line.length - node.innerText.length))
+
+      // update the lineId on the DOM for next use
+      node.dataset.lineId = newLine.getID()
+    },
+
+    /**
+     * @param {InputEvent} e
+     */
+     signsChanged(e, { line, node }) {
+
+      // synchronize the line > DOM
+      line.synchronizeTo(node.innerText)
+    },
+
+    /**
+     * @param {InputEvent} e
+     */
+    signsRemoved(e, { line, node }) {
+
+      // determine if the lines are the same as the count of <p> elements
+      // if not, then a line was deleted/merged into 
+      if (node.parentElement.children.length !== this.column.length) {
+        this.synchronize(node.parentElement)
+
+      } else {
+
+        // this was a simple removal of a sign, we can handle that
+        // via a signsChanged to diff the DOM
+        this.signsChanged(e, { line, node })
+      }
+    },
+
+    /**
+     * @param {KeyboardEvent} e triggered event
+     */
+    onCopy(e) {
+      // console.log('c', e)
+    },
+
+    /**
+     * @param {KeyboardEvent} e triggered event
+     */
+    onPaste(e) {
+      // console.log('p', e)
+    },
+
+    /**
+     * @param {InputEvent} e triggered event
+     */
+    onInput(e) {
+
+      switch (e.inputType) {
+
+        // inserted block/paragraph
+        case "insertParagraph":
+          this.insertLineAtSelection(e, this.getLineSelection())
+          break;
+
+        // inserted chars
+        case "insertText":
+          this.signsChanged(e, this.getLineSelection())
+          break
+
+        // deleted content. Could be line or single char
+        case "deleteContentBackward":
+          this.signsRemoved(e, this.getLineSelection())
+          break;
+
+        // for now, just log out what we're missing
+        default:
+          console.log('unhandle input event', e)
+      }
+    },
+
+    /**
+     * @returns {object} with this shape: {
+     *  line: {Line} the line model, where applicable
+     *  node: {HtmlElement} the DOM element corresponding to the line
+     * }
+     */
+    getLineSelection() {
+      let selectedNode = document.getSelection()
+
+      // safeguard to ensure a workable DOM element is available
+      if (!selectedNode || !selectedNode.anchorNode) {
+        return {}
+      }
+
+      return this.getLineParent(selectedNode.anchorNode)
```

Comment:
nit: I still don't love multiple returns in a single function (yes it stems my old CS professor scolding me).  I guess I would prefer a ternary:
return !selectedNode || !selectedNode.anchorNode
            ? {} 
            : this.getLineParent(selectedNode.anchorNode)

or something like that.

Reply:
Dang it. I've been lazy ;) I have a colleague at work who very much dislikes multiple returns as he was privy to a study awhile back that multiple returns leads to more bugs. 

In any case, I think the ternary is a good solution here. Happy to adopt the practice of no multiple returns. The only one I do think is ok is "safeguard" type early returns which check the validity of the params and bail if they're malformed.